tests/sentry/utils/test_event_frames.py

```diff
@@ -75,6 +75,137 @@ def test_platform_java_no_module(self):
+                "in_app": False,
+            },
+            {
+                "function": "dispatchMessage",
+                "module": "android.os.Handler",
+                "filename": "Handler.java",
+                "abs_path": "Handler.java",
+                "lineno": 99,
+                "in_app": False,
+            },
+            {
+                "function": "handleCallback",
+                "module": "android.os.Handler",
+                "filename": "Handler.java",
+                "abs_path": "Handler.java",
+                "lineno": 938,
+                "in_app": False,
+            },
+            {
+                "function": "run",
+                "module": "android.view.View$PerformClick",
+                "filename": "View.java",
+                "abs_path": "View.java",
+                "lineno": 28810,
+                "in_app": False,
+            },
+            {
+                "function": "access$3700",
+                "module": "android.view.View",
+                "filename": "View.java",
+                "abs_path": "View.java",
+                "lineno": 835,
+                "in_app": False,
+            },
+            {
+                "function": "performClickInternal",
+                "module": "android.view.View",
+                "filename": "View.java",
+                "abs_path": "View.java",
+                "lineno": 7432,
+                "in_app": False,
+            },
+            {
+                "function": "performClick",
+                "module": "com.google.android.material.button.MaterialButton",
+                "filename": "MaterialButton.java",
+                "abs_path": "MaterialButton.java",
+                "lineno": 1119,
+                "in_app": False,
+            },
+            {
+                "function": "performClick",
+                "module": "android.view.View",
+                "filename": "View.java",
+                "abs_path": "View.java",
+                "lineno": 7455,
+                "in_app": False,
+            },
+            {
+                "function": "onClick",
+                "module": "com.jetbrains.kmm.androidApp.MainActivity$$ExternalSyntheticLambda0",
+                "lineno": 2,
+                "in_app": True,
+            },
+            {
+                "function": "$r8$lambda$hGNRcN3pFcj8CSoYZBi9fT_AXd0",
+                "module": "com.jetbrains.kmm.androidApp.MainActivity",
+                "lineno": 0,
+                "in_app": True,
+            },
+            {
+                "function": "onCreate$lambda-1",
+                "module": "com.jetbrains.kmm.androidApp.MainActivity",
+                "filename": "MainActivity.kt",
+                "abs_path": "MainActivity.kt",
+                "lineno": 55,
+                "in_app": True,
+            },
+        ]
+        key, munged_frames = munged_filename_and_frames("java", exception_frames, "munged_filename")
+        assert len(munged_frames) == 16
```

Comment:
Could also be worth just checking that the `munged_filename` of the last frame is actually `com/jetbrains/kmm/androidApp/MainActivity.kt`